src/cbica_toolkit/src/cbicaUtilities.cpp

```diff
@@ -751,14 +749,16 @@ namespace cbica
     }
 #else
     //! Initialize pointers to file and user names
-    char path[PATH_MAX];
-    if (::readlink("/proc/self/exe", path, sizeof(path) - 1) == -1)
-      //path = dirname(path);
-      std::cerr << "[getFullPath()] Error during getting full path..";
+     char path[PATH_MAX];
+     if (::readlink("/proc/self/exe", path, sizeof(path) - 1) == -1)
+       std::cerr << "[getFullPath()] Error during getting full path..";
 #endif
 
-    std::string return_string = std::string(path);
-    path[0] = '\0';
+    if( return_string.empty())
+    {
+      return_string = std::string(path);
+      path[0] = '\0';
+    }
 
     return return_string;
   }
```

Comment:
- ```return_string``` is always empty, because it is never set before. The if statement is equal to ```if (true)```.
- Then you initialize ```return_string``` from ```path``` and then you modify the latter by ```path[0] = '\0';``` which is also useless because path is never used again. Right after the modification you return without using ```path``` again.

Reply:
You are right about the first point, which I have fixed. The second point is moot because I am basically setting that pointer to ```null```, ensuring there is no memory leak.